File: example-java/src/main/java/com/github/simonpercic/oklogexamplejava/DataFetcherComponent.java
package com.github.simonpercic.oklogexamplejava;

import com.github.simonpercic.oklogexamplejava.data.api.okhttp.DataModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
@Singleton
@Component(modules = DataModule.class)
public interface DataFetcherComponent {

    void inject(DataFetcher dataFetcher);
}

File: example-kotlin/src/main/kotlin/com/github/simonpercic/oklogexamplekotlin/DataFetcherComponent.kt
package com.github.simonpercic.oklogexamplekotlin

import com.github.simonpercic.oklogexamplekotlin.data.api.okhttp.DataModule

import javax.inject.Singleton

import dagger.Component

/**
 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
@Singleton
@Component(modules = arrayOf(DataModule::class))
interface DataFetcherComponent {

    fun inject(dataFetcher: DataFetcher)
}
